data/maps/SP_Scenarios/Along_The_River.wmf/scripting/functions.lua

```diff
@@ -0,0 +1,119 @@
+-- =======================================================================
+--                             Along The River
+-- =======================================================================
+
+-- Checks objective for done flag.
+function obj_done(objective)
+   return (objective ~= nil and objective.done)
+end
+
+-- =======================================================================
+
+-- Places building at field if player owns field and extra field for flag.
+function place_building_if_owner(player, building, point, cs)
+   if point.owner == player and point.brn.owner == player then
+      player:place_building(building, point, cs, true)
+      return true
+   end
+   return false
+end
+
+-- =======================================================================
+
+function remove_type(point, _type)
+   local immovable = point.immovable
+   if immovable then
+      if immovable.descr.type_name == _type then
+         immovable:remove()
+      end
+   end
+end
+
+-- =======================================================================
+
+function immovable_has_name(immovable, name)
+   return (immovable ~= nil and immovable.descr.name == name)
+end
+
+-- =======================================================================
+
+function add_wares(point, ware, amount)
+   local immovable = point.immovable
+   if immovable then
+      if immovable.descr.type_name == "flag" or immovable.descr.type_name == "warehouse" or immovable.descr.type_name == "productionsite" then
+         immovable:set_wares(ware, immovable:get_wares(ware) + amount)
+      end
+   end
+end
+
+-- =======================================================================
+
+-- Transports wares to flag, warehouse or productionsite.
+-- Waits if the building does not exist.
+function transport_wares(player, point, ware, amount, alert, image)
+   local count = amount
+   while count > 0 do
+      sleep(499)
+      local immovable = point.immovable
+      if immovable then
+         if immovable.descr.type_name == "flag" or immovable.descr.type_name == "warehouse" or immovable.descr.type_name == "productionsite" then
+            immovable:set_wares(ware, immovable:get_wares(ware) + 1)
+            count = count - 1
+         end
+      end
```

Comment:
How adding a boolean return value to `add_wares()` and reusing this function here?

Reply:
i don't think we need the boolean as the current design could result in an endless loop if the target immovable gets destroyed so I think using add wares and just count the attempts might do the trick.

Reply:
used add_wares now and simplified the usage a bit. removed the waiting part of the function to avoid endless loop. Removed the comment about this as well